src/libraries/System.Private.CoreLib/src/System/Guid.cs

```diff
@@ -130,14 +130,14 @@ public Guid(int a, short b, short c, byte[] d)
             _a = a;
             _b = b;
             _c = c;
-            _k = d[7]; // hoist bounds checks
```

Comment:
Using a relatively recent runtime, I don't see the bounds checks being removed if this line is moved back down. You do? @egorbo?

Reply:
Strange, as far back as .NET 6 (X64) I see just two branches, the first for the null check and the second for the length check.

Reply:
[sharplab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAZgAJZsATCAOwBsBPcgB2ygwEtsHzcMUAK5gM5AOJCu1ALAAoAN7zyKtlC4A3bBhiUYNes3Jc6YgPrYA3MtWt1WnXoOMWuABbRzwa3NVrN2rpUtC78HpzkZmA+fnYBjsGGLMBMjmbUMbb2gU4hRilpMJkqcQ5B+nnJqbpmAGbF/mW5SeQFNQDmDaU5iaFtkW5d2QkVLf1mXEPx5c751ZEAVlNNvXNpANY+Nipk5EJ0uNi1upLSABQmYthoYZ6tN+53YDdtANoAuuTUAJTb5Eq+PyqYgARgAdABBKDtIT4GCmAByQgYDAAoggwDBWNx6GCACpuKAQADuAElakiUWcfg0/H8/FxauRqWCADLw9oYNzkACEAF5yAAOX6AoH/eli8hconE8h0GCy0GQ6Gw+EYdGY7Fcehnb601QAXwlKmNkWw5AFVlNZmAFta+pUUTt0Xk1uoduorwADO8HZFdALPSDfdamYHXgAmEOioFmdoe16kaOSgYJlDJyUTBMAVgzYrMCwTSDzsfWCYA7CWjXJq/IgA===)

Reply:
Ah! I didn't see the length check up above. Ok, it makes sense now.

Reply:
if this method is perf sensitive, should the `throw new ArgumentException` be extracted?

Reply:
> if this method is perf sensitive

It's not. :)